### PR TITLE
add(board): Add M5Stack NanoC6 Dev Kit

### PR DESCRIPTION
**Pull request by lboue**
M5Stack NanoC6 Dev Kit
https://shop.m5stack.com/products/m5stack-nanoc6-dev-kit

*By completing this PR sufficiently, you help us to review this Pull Request quicker and also help improve the quality of Release Notes*

### Checklist
1. [x] Please provide specific title of the PR describing the change, including the component name (*eg. „Update of Documentation link on Readme.md“*)
2. [x] Please provide related links (*eg. Issue which will be closed by this Pull Request*)
3. [x] Please **update relevant Documentation** if applicable
4. [x] Please check [Contributing guide](https://docs.espressif.com/projects/arduino-esp32/en/latest/contributing.html)
5. [x] Please **confirm option to "Allow edits and access to secrets by maintainers"** when opening a Pull Request

*This entire section above can be deleted if all items are checked.*

-----------
## Description of Change
add(board): Add M5Stack NanoC6 Dev Kit

## Tests scenarios
Local testing with M5Stack NanoC6 Dev Kit.

## Related links

https://shop.m5stack.com/products/m5stack-nanoc6-dev-kit

(*eg. Closes #number of issue*)

### CODE REVIEW

File: variants/m5stack_stickc_plus/m5stack_nanoc6/pins_arduino.h [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+static const uint8_t SCL = 1;
+
+static const uint8_t SS = 21;
+static const uint8_t MOSI = 18;
+static const uint8_t MISO = 20;
+static const uint8_t SCK = 19;
+
+static const uint8_t A1 = 1;
+static const uint8_t A2 = 2;
+static const uint8_t A3 = 2;
```
Comment (me-no-dev) [inline variants/m5stack_stickc_plus/m5stack_nanoc6/pins_arduino.h:30]: same pin defined as A2 and A3

File: variants/m5stack_stickc_plus/m5stack_nanoc6/pins_arduino.h [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+
+static const uint8_t D1 = 1;
+static const uint8_t D2 = 2;
+static const uint8_t D3 = 3;
+static const uint8_t D4 = 7;
+static const uint8_t D5 = 8;
```
Comment (me-no-dev) [inline variants/m5stack_stickc_plus/m5stack_nanoc6/pins_arduino.h:36]: GPIO8 is not exposed

File: variants/m5stack_stickc_plus/m5stack_nanoc6/pins_arduino.h [hunk 3]
```diff
@@ -0,0 +1,45 @@
+#ifndef Pins_Arduino_h
```
Comment (me-no-dev) [inline variants/m5stack_stickc_plus/m5stack_nanoc6/pins_arduino.h:1]: Currently this file is in `variants/m5stack_stickc_plus/m5stack_nanoc6/pins_arduino.h` and should be in `variants/m5stack_nanoc6/pins_arduino.h`

**Comment by me-no-dev** [comment]
@P-R-O-C-H-Y PTAL if I missed anything

**Comment by me-no-dev** [comment]
@lboue would you please take care of the comments above?

**Comment by lboue** [comment]
I will do it ASAP

**Comment by lboue** [comment]
> I will do it ASAP

@me-no-dev Done

**Comment by me-no-dev** [comment]
@lboue thanks!

@P-R-O-C-H-Y PTAL

**Review by P-R-O-C-H-Y — APPROVED** [review]
LGTM

**Pull request merged**